src/julia-parser.scm

```diff
@@ -2285,10 +2285,14 @@
                       (if (closing-token? t)
                           (error (string "unexpected \"" (take-token s) "\"")))))
            (take-token s)
-           (if (and (eq? t 'var) (eqv? (peek-token s) #\") (not (ts:space? s)))
+           (if (and (eq? t 'var)
+                    (if (not (or (ts:pbtok s) (ts:last-tok s)))
+                      ;; Hack to preserve (io.pos (ts:port s)) for non-greedy Meta.parse
+                      (eqv? (peek-char (ts:port s)) #\"))
+                    (and (eqv? (peek-token s) #\") (not (ts:space? s))))
```

Comment:
I think it would be clearer to nest this inside the previous `if` statement.

Reply:
This ordering was intentional: we must run `peek-token` before the `take-token` further down.  However, we musn't run `peek-token` unless necessary.

It's delicate I'm afraid.

Perhaps I'll just add an extra `peek-token` down below and rearrange as suggested. That would be clearer if (very slightly) redundant.